scss/forms/_star-rating.scss

```diff
@@ -3,31 +3,33 @@
 //
 
 .star-rating {
-  &.disabled,
-  &:disabled {
-    pointer-events: none;
-  }
 
-  > label {
+  > label,
+  span {
     display: block;
     float: left;
     width: 1em;
     height: 1em;
     font-size: $form-star-size;
-    cursor: pointer;
-    @include form-star-rating($accessible-orange);
+    @include form-star-rating($accessible-orange, $accessible-orange);
 
     &:not(:first-of-type){
       margin-left: $form-star-margin-between;
     }
   }
 
-  > input:checked ~ label {
-    @include form-star-rating($gray-500);
+  > label {
+    cursor: pointer;
   }
 
-  > input:checked + label {
-    @include form-star-rating($accessible-orange);
+  > input:checked ~ label,
+  .checked ~ span {
```

Comment:
Maybe we should expand the readonly to the small variant and dark variant as well ?